language-antlr-utils/src/main/java/de/jplag/antlr/InternalListenerException.java

```diff
@@ -0,0 +1,15 @@
+package de.jplag.antlr;
+
+/**
+ * Exception type used internally within the antlr utils. Has to be a {@link RuntimeException}, because it is thrown
+ * within the antlr listener methods. Should not be thrown outside the antlr utils.
+ */
+public class InternalListenerException extends RuntimeException {
```

Comment:
The name of this exception does not really help with understand when this exception is thrown or what scenario it covers.